src/key_gen/mod.rs

```diff
@@ -245,9 +245,24 @@ impl<P: PublicId> AccumulateComplain<P> {
 
     fn finalize_complain(&self) -> BTreeSet<P> {
         let mut failings = BTreeSet::new();
+
+        // Counting for how many times a member missed complaining against others validly.
+        // If missed too many times, such member shall be considered as invalid directly.
+        let mut counting: BTreeMap<P, usize> = BTreeMap::new();
+
         for (target_id, accusers) in self.complains.iter() {
             if accusers.len() * 3 > self.pub_keys.len() * 2 {
```

Comment:
Why 2/3 here and not >50%? 2/3 seems like overkill, bu tthe quorum/threshold should be set by users. Just looking fo rmore info here.

Reply:
It's not mentioned `explicitly` of the criteria that how many votes shall be collected when consider a peer is valid. as it is expected that `ALL` member shall vote,

Reply:
Should we though? I mean if a node never voted then it's bad. So should we not either complain or ask for it to resend to us? Or ask another ndoe to send us the missing vote?

Reply:
Maybe a TODO? Where we accept 2/3 for now? but if somebody chooses a threshold larger than 2/3 what happens?

Reply:
> So should we not either complain or ask for it to resend to us? Or ask another ndoe to send us the missing vote?
This is within the `finalize_complaint`, with the the `completion of the Complaint phase`. So the calculation just let us to decide whether shall the invalid candidate be removed from out pub_keys (expectation list). There is no more complain or ask for resend or ask another node for the missing vote.

Reply:
OK Cheers @maqi  Can I ask why not use the threshold for this then? It seems that is the minimum required,  perhaps I am reading this wrong?

Reply:
right, for the case of when a threshold larger than 2/3, then it may have to use the threshold instead of 2/3.
I will change it then.